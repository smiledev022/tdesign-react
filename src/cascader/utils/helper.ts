import { TreeNode } from '../interface';

/**
 * 获取node节点的全路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}
