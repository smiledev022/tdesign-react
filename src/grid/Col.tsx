import React from 'react';
import classNames from 'classnames';
import isObject from 'lodash/isObject';
import useConfig from '../_util/useConfig';
import { StyledProps } from '../_type';

type FlexType = number | 'none' | 'auto' | string;

/**
 * Divider 组件支持的属性。
 */
export interface ColProps extends StyledProps {
  /**
   * flex 布局填充
   *
   *  @default
   */
  flex?: FlexType;

  /**
   * 栅格左侧的间隔格数，间隔内不可以有栅格
   *
   * @default 0
   */
  offset?: number;

  /**
   * 栅格向左移动格数
   *
   * @default 0
   */
  pull?: number;

  /**
   * 栅格向右移动格数
   *
   * @default 0
   */
  push?: number;

  /**
   * 栅格所占位格数，为 0 时相当于 display: none
   *
   * @default 0
   */
  span?: number;

  /**
   * 栅格所占位格数，为 0 时相当于 display: none
   *
   * @default 0
   */
  order?: number;

  /**
   * 自定义元素标签
   *
   * @default div
   */
  tag?: string;

  /**
   * 响应式栅格
   *
   */
  xs?: number | number[];
  sm?: number | number[];
  md?: number | number[];
  lg?: number | number[];
  xl?: number | number[];
  xxl?: number | number[];

  /**
   * 文本内容
   */
  children?: React.ReactNode;
}

const calcColPadding = (gutter: any, currentSize: string) => {
  let padding = '';
  if (typeof gutter === 'number') {
    padding = `0 ${gutter / 2}px`;
  } else if (Array.isArray(gutter) && gutter.length) {
    padding = `0 ${gutter[0] / 2}px`;
  } else if (isObject(gutter) && gutter[currentSize]) {
    if (Array.isArray(gutter[currentSize])) {
      padding = `0 ${gutter[currentSize][0] / 2}px`;
    } else {
      padding = `0 ${gutter[currentSize] / 2}px`;
    }
  }
  return padding;
};

const parseFlex = (flex: FlexType) => {
  if (typeof flex === 'number') {
    return `${flex} ${flex} auto`;
  }

  if (/^\d+(\.\d+)?(px|em|rem|%)$/.test(flex)) {
    return `0 0 ${flex}`;
  }

  return flex;
};

/**
 * Col组件
 */
const Col = (props: ColProps | any) => {
  const {
    flex,
    offset,
    order,
    pull,
    push,
    span,
    tag = 'div',
    className,
    children,
    style: propStyle,
    ...otherColProps
  } = props;
  const { gutter: rowGutter, size: rowSize } = otherColProps;

  const { classPrefix } = useConfig();
  const allSizes = ['xs', 'sm', 'md', 'lg', 'xl', 'xxl'];
  const sizeClassObj = allSizes.reduce((acc, currSize) => {
    const propSize = props[currSize];
    let sizeProps: any = {};
    if (typeof propSize === 'number') {
      sizeProps.span = propSize;
    } else if (isObject(propSize)) {
      sizeProps = propSize || {};
    }

    delete otherColProps[currSize];

    return {
      ...acc,
      [`${classPrefix}-col-${currSize}-${sizeProps.span}`]: sizeProps.span !== undefined,
      [`${classPrefix}-col-${currSize}-order-${sizeProps.order}`]: sizeProps.order !== undefined,
      [`${classPrefix}-col-${currSize}-offset-${sizeProps.offset}`]: sizeProps.offset !== undefined,
      [`${classPrefix}-col-${currSize}-push-${sizeProps.push}`]: sizeProps.push !== undefined,
      [`${classPrefix}-col-${currSize}-pull-${sizeProps.pull}`]: sizeProps.pull !== undefined,
    };
  }, {});

  const colClassNames = classNames(
    `${classPrefix}-col`,
    className,
    {
      [`${classPrefix}-col-${span}`]: span !== undefined,
      [`${classPrefix}-col-offset-${offset}`]: offset !== undefined,
      [`${classPrefix}-col-pull-${pull}`]: pull !== undefined,
      [`${classPrefix}-col-push-${push}`]: push !== undefined,
      [`${classPrefix}-col-order-${order}`]: order !== undefined,
    },
    sizeClassObj,
  );

  const ColStyle = {
    padding: calcColPadding(rowGutter, rowSize),
    ...propStyle,
  };
  flex && (ColStyle.flex = parseFlex(flex));

  return React.createElement(
    tag,
    {
      className: colClassNames,
      style: ColStyle,
      ...otherColProps,
    },
    children,
  );
};

Col.displayName = 'Col';

export default Col;
