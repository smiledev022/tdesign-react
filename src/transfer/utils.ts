import isEmpty from 'lodash/isEmpty';
import isFunction from 'lodash/isFunction';
import isString from 'lodash/isString';
import React from 'react';
import { TNode } from '../common';

export const getDefaultValue = <T>(value: T | T[], defaultValue?: T): T[] => {
  if (isEmpty(value)) {
    return Array.isArray(defaultValue) ? defaultValue : [defaultValue, defaultValue];
  }

  if (Array.isArray(value)) return value;

  return [defaultValue, defaultValue];
};

export const getJSX = (value: string | TNode) => {
  if (isString(value)) return value;

  if (isFunction(value)) return value();

  if (React.isValidElement(value)) return value;

  return value;
};
