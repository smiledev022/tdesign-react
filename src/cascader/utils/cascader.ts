import { getFullPathLabel } from './helper';
import { TreeNode, TreeNodeValue, TreeOptionData, CascaderContextType } from '../interface';

/**
 * treeValue计算方法
 * @param model
 * @returns
 */
export const getTreeValue = (model: CascaderContextType['model']) => {
  let treeValue: TreeNodeValue[] = [];
  if (Array.isArray(model)) {
    if (model.length > 0 && typeof model[0] === 'object') {
      treeValue = (model as TreeOptionData[]).map((val) => val.value);
    } else if (model.length) {
      treeValue = model as TreeNodeValue[];
    }
  } else if (model) {
    if (typeof model === 'object') {
      treeValue = [(model as TreeOptionData).value];
    } else {
      treeValue = [model];
    }
  }
  return treeValue;
};

/**
 * input和treeStore变化的副作用
 * @param inputVal
 * @param treeStore
 * @param setTreeNodes
 * @returns
 */
export const treeNodesEffect = (
  inputVal: CascaderContextType['inputVal'],
  treeStore: CascaderContextType['treeStore'],
  setTreeNodes: CascaderContextType['setTreeNodes'],
) => {
  if (!treeStore) return;
  let nodes = [];
  if (inputVal) {
    nodes = treeStore.nodes.filter((node: TreeNode) => {
      const fullPathLabel = getFullPathLabel(node);
      return fullPathLabel.indexOf(inputVal) > -1 && node.isLeaf();
    });
  } else {
    nodes = treeStore.getNodes().filter((node: TreeNode) => node.visible);
  }
  setTreeNodes(nodes);
};

/**
 * 初始化展开阶段与展开状态副作用
 * @param treeStore
 * @param treeValue
 * @param expend
 */
export const treeStoreExpendEffect = (
  treeStore: CascaderContextType['treeStore'],
  treeValue: TreeNodeValue[],
  expend: TreeNodeValue[],
) => {
  // init expanded, 无expend状态时设置
  if (Array.isArray(treeValue) && expend.length === 0) {
    const expandedMap = new Map();
    const [val] = treeValue;
    if (val) {
      expandedMap.set(val, true);
      const node = treeStore.getNode(val);
      if (!node) return;
      node.getParents().forEach((tn: TreeNode) => {
        expandedMap.set(tn.value, true);
      });
      const expandedArr = Array.from(expandedMap.keys());
      treeStore.replaceExpanded(expandedArr);
    }
  }
  // 本地维护 expend，更加可控，不需要依赖于 tree 的状态
  if (treeStore.getExpanded() && expend.length) {
    treeStore.replaceExpanded(expend);
  }
  treeStore.refreshNodes();
};
