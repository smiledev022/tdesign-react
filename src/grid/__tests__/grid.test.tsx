import React from 'react';
import { testExamples, render } from '@test/utils';
import Row from '../Row';
import Col from '../Col';

// 测试组件代码 Example 快照
testExamples(__dirname);

describe('Grid 组件测试', () => {
  test('Grid ', async () => {
    const { container } = render(
      <Row>
        <Col></Col>
      </Row>,
    );

    // 校验默认className
    const defaultClass = ['t-row'];
    expect(container.firstChild).toHaveClass(...defaultClass);
  });
});
