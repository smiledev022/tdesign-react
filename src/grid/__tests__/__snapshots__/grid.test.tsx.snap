// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignGrid.jsx 1`] = `
<DocumentFragment>
  <p>
    align top
  </p>
  <div
    class="t-row t-row-center t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 80px;"
      >
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 40px;"
      >
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 80px;"
      >
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 40px;"
      >
        col-4
      </div>
    </div>
  </div>
  <p>
    Align Middle
  </p>
  <div
    class="t-row t-row-space-around t-row-middle"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 80px;"
      >
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 40px;"
      >
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 80px;"
      >
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 40px;"
      >
        col-4
      </div>
    </div>
  </div>
  <p>
    Align Bottom
  </p>
  <div
    class="t-row t-row-space-between t-row-bottom"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 80px;"
      >
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 40px;"
      >
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 80px;"
      >
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div
        style="height: 40px;"
      >
        col-4
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BasicGrid.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
    <div
      class="t-col t-col-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
    <div
      class="t-col t-col-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-3"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        3
      </div>
    </div>
    <div
      class="t-col t-col-3"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        3
      </div>
    </div>
    <div
      class="t-col t-col-3"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        3
      </div>
    </div>
    <div
      class="t-col t-col-3"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        3
      </div>
    </div>
    <div
      class="t-col t-col-3"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        3
      </div>
    </div>
    <div
      class="t-col t-col-3"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        3
      </div>
    </div>
    <div
      class="t-col t-col-3"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        3
      </div>
    </div>
    <div
      class="t-col t-col-3"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        3
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        4
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-6"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        6
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-8"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        8
      </div>
    </div>
    <div
      class="t-col t-col-8"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        8
      </div>
    </div>
    <div
      class="t-col t-col-8"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        8
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-12"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        12
      </div>
    </div>
    <div
      class="t-col t-col-12"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        12
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-24"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        24
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DisplayGrid.jsx 1`] = `
<DocumentFragment>
  <p>
    align left
  </p>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
  </div>
  <p>
    align center
  </p>
  <div
    class="t-row t-row-center t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
  </div>
  <p>
    align right
  </p>
  <div
    class="t-row t-row-end t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
  </div>
  <p>
    space-between
  </p>
  <div
    class="t-row t-row-space-between t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
  </div>
  <p>
    space-around
  </p>
  <div
    class="t-row t-row-space-around t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
    <div
      class="t-col t-col-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-4
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FlexGrid.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col"
      gutter="0"
      style="padding: 0px 0px; flex: 2 2 auto;"
    >
      <div>
        2 / 5
      </div>
    </div>
    <div
      class="t-col"
      gutter="0"
      style="padding: 0px 0px; flex: 3 3 auto;"
    >
      <div>
        3 / 5
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col"
      gutter="0"
      style="padding: 0px 0px; flex: 0 0 100px;"
    >
      <div>
        100px
      </div>
    </div>
    <div
      class="t-col"
      gutter="0"
      style="padding: 0px 0px; flex-basis: auto;"
    >
      <div>
        Fill Rest
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col"
      gutter="0"
      style="padding: 0px 0px; flex: 1 1 200px;"
    >
      <div>
        1 1 200px
      </div>
    </div>
    <div
      class="t-col"
      gutter="0"
      style="padding: 0px 0px; flex: 0 1 300px;"
    >
      <div>
        0 1 300px
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col"
      gutter="0"
      style="padding: 0px 0px; flex: 0 0 auto;"
    >
      <div
        style="padding: 0px 16px;"
      >
        none
      </div>
    </div>
    <div
      class="t-col"
      gutter="0"
      style="padding: 0px 0px; flex-basis: auto;"
    >
      <div>
        auto with no-wrap
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`OffsetGrid.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-8"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-8
      </div>
    </div>
    <div
      class="t-col t-col-8 t-col-offset-8"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-8
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-6 t-col-offset-6"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-6 col-offset-6
      </div>
    </div>
    <div
      class="t-col t-col-6 t-col-offset-6"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-6 col-offset-6
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-12 t-col-offset-6"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        col-12 col-offset-6
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`OrderGrid.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-6 t-col-order-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1 col-order-4
      </div>
    </div>
    <div
      class="t-col t-col-6 t-col-order-3"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2 col-order-3
      </div>
    </div>
    <div
      class="t-col t-col-6 t-col-order-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        3 col-order-2
      </div>
    </div>
    <div
      class="t-col t-col-6 t-col-order-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        4 col-order-1
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-6 t-col-xs-order-1 t-col-sm-order-2 t-col-md-order-3 t-col-lg-order-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        1 col-order-responsive
      </div>
    </div>
    <div
      class="t-col t-col-6 t-col-xs-order-2 t-col-sm-order-1 t-col-md-order-4 t-col-lg-order-3"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        2 col-order-responsive
      </div>
    </div>
    <div
      class="t-col t-col-6 t-col-xs-order-3 t-col-sm-order-4 t-col-md-order-2 t-col-lg-order-1"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        3 col-order-responsive
      </div>
    </div>
    <div
      class="t-col t-col-6 t-col-xs-order-4 t-col-sm-order-3 t-col-md-order-1 t-col-lg-order-2"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        4 col-order-responsive
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ResponseGrid.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -0px;"
  >
    <div
      class="t-col t-col-xs-2 t-col-sm-4 t-col-md-6 t-col-lg-8 t-col-xl-10"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        Col
      </div>
    </div>
    <div
      class="t-col t-col-xs-20 t-col-sm-16 t-col-md-12 t-col-lg-8 t-col-xl-4"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        Col
      </div>
    </div>
    <div
      class="t-col t-col-xs-2 t-col-sm-4 t-col-md-6 t-col-lg-8 t-col-xl-10"
      gutter="0"
      style="padding: 0px 0px;"
    >
      <div>
        Col
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SpaceGrid.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -8px;"
  >
    <div
      class="t-col t-col-6"
      gutter="16"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="16"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="16"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="16"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -12px;"
  >
    <div
      class="t-col t-col-6"
      gutter="[object Object]"
      style="padding: 0px 12px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="[object Object]"
      style="padding: 0px 12px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="[object Object]"
      style="padding: 0px 12px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="[object Object]"
      style="padding: 0px 12px;"
    >
      <div>
        col-6
      </div>
    </div>
  </div>
  <div
    class="t-row t-row-start t-row-top"
    style="margin: 0px -8px;"
  >
    <div
      class="t-col t-col-6"
      gutter="16,24"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="16,24"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="16,24"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="16,24"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="16,24"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="16,24"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="16,24"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
    <div
      class="t-col t-col-6"
      gutter="16,24"
      style="padding: 0px 8px;"
    >
      <div>
        col-6
      </div>
    </div>
  </div>
</DocumentFragment>
`;
